iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NodeNumberBasedBusRef.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.powsybl.commons.PowsyblException;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * @author Yichen TANG <yichen.tang at rte-france.com>
+ */
+public class NodeNumberBasedBusRef extends AbstractBusRef {
+
+    private final String voltageLevelId;
+    private final int node;
+
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+    public NodeNumberBasedBusRef(@JsonProperty("voltageLevelId") String voltageLevelId, @JsonProperty("node") int node) {
+        this.voltageLevelId = Objects.requireNonNull(voltageLevelId);
+        this.node = node;
+    }
+
+    @Override
+    Optional<Bus> resolveByLevel(Network network, TopologyLevel level) {
+        final VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelId);
+        if (voltageLevel == null) {
+            return Optional.empty();
+        }
+        if (Objects.equals(TopologyKind.NODE_BREAKER, voltageLevel.getTopologyKind())) {
+            final Terminal terminal = voltageLevel.getNodeBreakerView().getTerminal(node);
```

Comment:
A node is not guaranteed to have an associated terminal, so this will not work as intended.

We would need a way to retrive the bus for this node, but the API does not allow it ... As it will not be a need for now I propose to simply delete that implementation (yes it leaves us with only one implementation for now).